### PR TITLE
[4.3] fix for #40182 : Router error in NomenuRules

### LINKED ISSUE #40182: [4.3 RC1] : NoMenuRules.php : undefined $this->app
### Expected result
No Error

### Actual result
ERROR 0 - Call to a member function getInput() on null in ...\libraries\src\Component\Router\Rules\NomenuRules.php:82
$this->app is not initialized

### System information (as much as possible)
Joomla 4.3 RC1
PHP 8.1

### Additional comments
I found this issue with PhocaDownload extension, but, I don't think it's related to this extension.
In version 4.2.9, $this->router->app is being used instead of $this->app.

### PR DESCRIPTION
**Pull request by conseilgouz**
Pull Request for Issue #40182 .

### Summary of Changes
In version 4.2.9, $this->router->app is being used instead of $this->app.

### Testing Instructions
Use Joomla 4.3 nightly build
Create at least one article
Create an instance of Articles - Latest module
Change Home menu item to link to Featured Contacts menu item type. If you have any menu items link to com_content before, please unpublish it or make them 'hidden'
Try to access to an article from Articles - Latest module above

### Actual result BEFORE applying this Pull Request

ERROR 0 - Call to a member function getInput() on null in ...\libraries\src\Component\Router\Rules\NomenuRules.php:82

### Expected result AFTER applying this Pull Request

No error

### CODE REVIEW

File: libraries/src/Component/Router/Rules/NomenuRules.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function parse(&$segments, &$vars)
 
                 if (isset($view->key) && isset($segments[0])) {
                     if (\is_callable([$this->router, 'get' . ucfirst($view->name) . 'Id'])) {
-                        $input = $this->app->getInput();
+                        $input = $this->router->app;
```
Comment (joomdonation) [inline libraries/src/Component/Router/Rules/NomenuRules.php]: There is still typo here. Please change it to `$input = $this->router->app->getInput();`

**Comment by joomdonation** [comment]
### Testing Instructions
1. Use Joomla 4.3 nightly build 
2. Create at least one article
3. Create an instance of Articles - Latest module
4. Change Home menu item to link to Featured Contacts menu item type. If you have any menu items link to com_content before, please unpublish it.
5. Try to access to an article from Articles - Latest module above


### Actual result BEFORE applying this Pull Request
You get fatal error: ERROR 0 - Call to a member function getInput() on null in ...\libraries\src\Component\Router\Rules\NomenuRules.php:82
$this->app is not initialized


### Expected result AFTER applying this Pull Request
No error. The article is being displayed properly.

**Comment by conseilgouz** [comment]
No, bug has been introduced by #39029
So, input() lines have to become getInput().

**Comment by conseilgouz** [comment]
Just updating line 82 causes line 83 crash.

**Comment by conseilgouz** [comment]
> Line 83 crash because you also made change to that line. If you just make change to line 82 as I mentioned, it will work well

No, original line contains if ($view->parent_key && $input->get($view->parent_key)) {

If I use your line, it crashes as $input is a method and not an object.

**Comment by conseilgouz** [comment]
Lines 83 to 85 have been updated to be consistent with pr 39029

**Comment by joomdonation** [comment]
> If I use your line, it crashes as $input is a method and not an object.

$input is an object. Not method. Like 82, if you make change as mentioned, will return application input object

```php
$input = $this->router->app->getInput();
```
`$this->router->app` returns the application object. And calling `$this->router->app->getInput();` will return input object, not a method. You can follow the testing instructions which provided to test the result yourself.

**Comment by conseilgouz** [comment]
I tried PR as we discussed : 
1. just updating line 82 to $input = $this->router->app; works, with no other change (but it is not consistent with PR #39029 )
2. Updating line 82 as you suggested causes an error on line 83

**Comment by joomdonation** [comment]
Your code does not work. I applied the changes (which I know it's wrong) and see some notices and 404 not found error:

Notice
: Object of class Joomla\CMS\Input\Input could not be converted to float in
libraries\vendor\joomla\database\src\Mysqli\MysqliStatement.php
on line
435

Notice
: Object of class Joomla\CMS\Input\Input could not be converted to int in
\libraries\vendor\joomla\database\src\Mysqli\MysqliStatement.php
on line
435

**Comment by conseilgouz** [comment]
I just tried following your testing instructions and, indeed, your single line works.
But, when I try with Phocadownload component, it blows up. 
May be a phocadownload issue.
Im doing some more research on this.

**Comment by conseilgouz** [comment]
For information, hidden components behave like no-menu components
After some more testing, I was not able to reproduce error on line 83 with other components (including phocadownload). I must have dreamt it...
So this PR only contains one line.

**Comment by sdwjoomla** [comment]
I have tested this item :white_check_mark: successfully on 6c948fa91740634bf862033171e8ed5de3da59c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40184">issues.joomla.org/tracker/joomla-cms/40184</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40184">issues.joomla.org/tracker/joomla-cms/40184</a>.</sub>

**Pull request merged**